src/gate/parser/bench/parser.cpp

```diff
@@ -0,0 +1,295 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the Utopia EDA Project, under the Apache License v2.0
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2021 ISP RAS (http://www.ispras.ru)
+//
+//===----------------------------------------------------------------------===//
+
+#include <FlexLexer.h>
+#include "gate/model/gnet.h"
+#include "gate/model/gate.h"
+#include "gate/model/gsymbol.h"
+#include "headerFile"
+#include "tokens.h"
+
+#include <iostream>
+#include <stdlib.h>
+#include <vector>
+#include <map>
+
+extern "C" int scan_token();
+
+///Defines the compare method for TokenMap.name.
+#define COMPARE(text) do {\
+  for (int j = 0; j < strlen(yytext); j++) {\
+    if (i.name[j] == yytext[j])\
+      k += 1;\
+  }\
+  if (k == strlen(yytext)) {\
+    std::cout << "ERROR IN " << text << " WITH "\
+      << yytext << "\tline: " << line << std::endl;\
+    exit(EXIT_FAILURE);\
+  }\
+} while (false)
+
+///Defines the alert method for debugging information.
+#define ALERT(errorString) do {\
+  std::cout << " expected '" << errorString << "'\tline: "\
+    << line << "\tСaught " << yytext << std::endl;\
+  exit(EXIT_FAILURE);\
+} while (false) 
+
+///Defines the method for verifying the next token.
+#define ASSERT_NEXT_TOKEN(expectedToken, errorString) do {\
+  Tokens token = getNextToken();\
+  if ( token != expectedToken ) {\
+    ALERT(errorString);\
+  }\
+} while (false)
+
+///Defines the adding in a token map vector.
+#define MAPS(definite, typeInit) do {\
+  char* text = new char[strlen(yytext)];\
+  strcpy(text, yytext);\
+  TokenMap map {text, definite, typeInit};\
+  maps.push_back(map);\
```

Comment:
- memory allocated and never deleted
- not enough memory allocated: C-strings are 0-terminated, therefore 1 more character required